source/grid/grid_generator.cc

```diff
@@ -4514,43 +4514,43 @@ namespace GridGenerator
     Assert(slice_coordinates.size() >= 2,
            ExcMessage(
              "The number of slices for extrusion must be at least 2."));
-    const unsigned int n_slices = slice_coordinates.size();
     Assert(std::is_sorted(slice_coordinates.begin(), slice_coordinates.end()),
            ExcMessage("Slice z-coordinates should be in ascending order"));
+    const std::size_t        n_slices = slice_coordinates.size();
```

Comment:
Like you mentioned below, it won't ever really matter, but I have tried to make a habit of doing this to avoid warnings in the compiler about narrowing. I'll fix the part below.